### PR TITLE
Round object

### PR DESCRIPTION
**Pull request by BRHoward**
Taken out a lot of the round data into it's own object.
Fixed replace cards bug which removes leaderboard data for everyone.
Fixed bug where on refresh sidebars dissappear.
Consolidated a lot of the answer and results data being sent in to one roundSubmissionData object.
Fixed bug where bots didn't submit answers when the timer runs out on the question page.

### CODE REVIEW

**Comment by JamesAdamHughes** [comment]
rebase plz

File: client/Angular/services/gameService.js [hunk 1]
```diff
@@ -118,15 +130,15 @@ ClonageApp.service('gameService', ['communicationService', 'dynamicTextService',
...
-		if (votes.length > 0) {
-			votes.forEach(function(vote) {
-				if (vote.playersWhoVotedForThis.length > 0) {
-					vote.playersWhoVotedForThis.forEach(function(player) {
-						if (player.name === user)
+		if (currentNumberOfVotes > 0) {
+			roundSubmissionData.forEach(function(submission) {
+				if (submission.playersWhoVotedForThis.length > 0) {
+					submission.playersWhoVotedForThis.forEach(function(player) {
+						if (player.name === userName)
```
Comment (JamesAdamHughes) [inline client/Angular/services/gameService.js]: can we change this now to be player.uid, not player name

File: client/Angular/services/gameService.js [hunk 2]
```diff
@@ -179,33 +191,32 @@ ClonageApp.service('gameService', ['communicationService', 'dynamicTextService',
...
+
+		roundSubmissionData = data.roundSubmissionData;
+		currentNumberOfSubmissions = data. currentNumberOfSubmissions;
+		currentNumberOfVotes = data.currentNumberOfVotes;
 
+		if(data.dontResetAnswers === undefined){
+			answersToVisualise = data.roundSubmissionData;
+		}
+
+		console.log("currentVOTES" + currentNumberOfVotes);
```
Comment (JamesAdamHughes) [inline client/Angular/services/gameService.js]: console log

File: server/GameController.js [hunk 3]
```diff
@@ -206,8 +164,11 @@ module.exports = function(data) {
 
 			data = {
 				players: players,
-				roundQuestion: round.question,
-				round: roundCount,
+				roundSubmissionData: round.getRoundSubmissionData(),
```
Comment (JamesAdamHughes) [inline server/GameController.js]: can you just return the round object? Or have a round to Object method that returns all neccessary fields?

File: server/GameController.js [hunk 4]
```diff
@@ -304,103 +257,39 @@ module.exports = function(data) {
 		var results = [];
 		var submittingPlayer = getPlayerFromId(playerId);
 
-		currentRound.results = [];
+		// currentRound.results = [];
```
Comment (JamesAdamHughes) [inline server/GameController.js]: comment

File: server/server.js [hunk 5]
```diff
@@ -603,9 +608,10 @@ module.exports = function(port, enableLogging, testing) {
                 });
 
 
-                room.broadcastRoom("GAME playerRoundResults", {
-                    results: data.results,
-                    voteCounter: data.voteCounter
+                room.broadcastRoom("GAME roundSubmissionData", {
+                    roundSubmissionData: data.roundSubmissionData,
```
Comment (JamesAdamHughes) [inline server/server.js]: lots of duplication of 
```
    roundSubmissionData: data.roundSubmissionData,
    currentNumberOfVotes: data.currentNumberOfVotes,
    currentNumberOfSubmissions: data.currentNumberOfSubmissions,
    dontResetAnswers: true
```

Comment (JamesAdamHughes) [inline server/server.js]: maybe can just send the data and extract it on the otherside? up for discussion

File: client/Angular/services/gameService.js (hunk 2 above)
Comment (Geor23) [inline client/Angular/services/gameService.js]: console log

**Comment by JamesAdamHughes** [comment]
jesus fuck

File: client/Angular/services/gameService.js [hunk 6]
```diff
@@ -179,33 +191,32 @@ ClonageApp.service('gameService', ['communicationService', 'dynamicTextService',
...
 		//generating the filled in question based on the question text and submitted answers as above
 		if (currentQuestion !== undefined) {
-			answers.forEach(function(answer) {
-				answer.filledInText = dynamicTextService.fillInSelections(currentQuestion.text, answer.answersText);
+			roundSubmissionData.forEach(function(submission) {
+				submission.filledInText = dynamicTextService.fillInSelections(currentQuestion.text, submission.submissionsText);
 			});
 		}
 
-		countdown = data.countdown;
```
Comment (Geor23) [inline client/Angular/services/gameService.js]: why is this removed? is countdown value set somewhere else ?

Comment (JamesAdamHughes) [inline client/Angular/services/gameService.js]: yea you jerk

File: test/unit/frontend/karma.gameService.spec.js [hunk 7]
```diff
@@ -65,23 +65,6 @@ describe("Testing Game Service", function() {
 		expect(gameService.getCurrentQuestion().pick).toEqual(1);
 	});
```
Comment (Geor23) [inline test/unit/frontend/karma.gameService.spec.js]: no round functions tests?

File: client/Angular/services/gameService.js (hunk 6 above)
Comment (BRHoward) [inline client/Angular/services/gameService.js]: countdown is set in the receive question function which is also called on refresh so this shouldn't make a difference to timer behaviour

File: server/server.js (hunk 5 above)
Comment (BRHoward) [inline server/server.js]: think i prefer the explicitness of this method though
